application-ownership-models/README.md

```diff
@@ -0,0 +1,116 @@
+
+Additionally, while application lifecycle stages split the app ownership between external (upstream) and internal (Giant Swarm) teams, sole or shared ownership is about how we handle app ownership internally within Giant Swarm.
+
+### 2.1. Sole application ownership
+
+The is the classical and usually the obvious case. A Giant Swarm team develops the application, in case of full application lifecycle ownership described in the previous section, or the team takes care of release engineering and operations for the app, in case of partial application lifecycle ownership described above.
+
+### 2.2. Shared application ownership
+
+And this is the tricky part, one of the main reasons for which this whole RFC is written.
+
+Shared application ownership applies in some cases of partial application lifecycle ownership, so in a nutshell, the question here is - which Giant Swarm team owns some upstream application? Can it even be one team, or multiple teams must share ownership? If multiple teams share ownership, how does that work in practice?
+
+There is no silver bullet for all upstream applications, so here we need clear and robust, but also flexible ownership model.
+
+Before trying to come up with some definition, let's first check few examples.
+
+#### 2.2.1. Examples of shared applications
+
+While there are more examples, let's take the following two that should paint the picture and describe the issue of shared ownership clearly enough.
+
+**Cluster API provider-independent controllers**
+
+There are three CAPI controllers that are equally used by KaaS teams (Rocket, Hydra, Clippy) - core, kubeadm control plane and kubeadm bootstrap controller. They are all packaged together in the cluster-api-app. All KaaS teams have equal stakes in this app and equal responsibilities.
+
+Which teams is responsible for maintaining the application repository, improving it, automating it, taking care of renovate and other automated pull requests? Which teams takes care of upgrading the app to the latest upstream? Every team for themselves when they have a need? This is not working very efficiently.
+
+**Kubernetes components**
+
+We could say that deploying and using Kubernetes is sort of what we all do for living, right?
+
+Quick reminder that here we are talking about partial application lifecycle ownership, so release engineering and operations, where we can't really say that we do former one, so the latter one is interesting here.
+
+All teams write or package operators that get scheduled to some node, operators that use and maybe sometimes hammer API server, deploy few or few hundreds of CRDs that case clusters to be overloaded, etc.
```

Comment:
IMO here we need better visibility. If we cannot identify which of the components is hammering down the API, we would not know which team is responsible

Reply:
Agreed definitely, we need better visibility there. This is just an example, I believe Ludacris had, and whichever team gets to own "core" k8s components will have similar challenges, which is why I added this as an example, just for the sake of painting a wider picture in the RFC, to give different examples where it could help (not to have those specific examples as proposals in the RFC).